File: Insert_In_Sorted_Order.java
public void insertInSortedOrder(Node[] head, int data) {
    Node newNode = new Node(data);
    if (head[0] == null || data < head[0].data) {
        insertFront(head, newNode);
        return;
    }
    Node current = head[0];
    while (current.link != null && data > current.link.data) {
        current = current.link;
    }
    newNode.link = current.link;
    current.link = newNode;
}

File: Insert_In_Sorted_Order.c
void insertInSortedOrder(NODE* head, int data) {
    NODE newNode = createNode(data);
    if (*head == NULL || data < (*head)->data) {
        insertFront(head, newNode);
        return;
    }
    NODE current = *head;
    while (current->link != NULL && data > current->link->data) {
        current = current->link;
    }
    newNode->link = current->link;
    current->link = newNode;
}
